syntax = "proto3";

service CredentialService {
  rpc Create(CheckedCredential) returns (Empty);
  rpc GetById (CredentialId) returns (Credential);
  rpc GetByEmail (CredentialEmail) returns (Credential);
  rpc Update (Credential) returns (Empty);
  rpc UpdatePassword (CheckedCredential) returns (Empty);
  rpc UpdateEmail (CheckedCredential) returns (Empty);
  rpc Deactivate (CredentialEmail) returns (Empty);
}

message Empty {
}

message Credential {
  string id = 1;
  string email = 2;
  string password = 3;
}

message CheckedCredential {
  string email = 1;
  string password = 2;
  string password_check = 3;
}

message CredentialId {
  string id = 1;
}

message CredentialEmail {
  string email = 1;
}